kernel.py

```diff
@@ -79,11 +80,8 @@ def __init__(self, connection_filename="kernel.json", connection_fn_with_pid=Tru
         self._lock = threading.Lock()
         self._condition = threading.Condition(lock=self._lock)
 
-        if connection_fn_with_pid:
-            name, ext = os.path.splitext(connection_filename)
-            connection_filename = "%s-%i%s" % (name, os.getpid(), ext)
-        self.connection_filename = connection_filename
-
+        self.connection_filename, self._should_reduce_filename = self._craft_connection_filename(
+            connection_filename, connection_fn_with_pid)
```

Comment:
Maybe better directly also inline this? I think it becomes more readable then.

Reply:
I did not do so because I prefer not to charge the `__init__` method as every reader must start there. Maybe they don't care about connection file creation (15 lines).

Reply:
Ok.